app/upgrades.go

```diff
@@ -120,3 +126,71 @@ func upgradeHandler(
 		return toVM, nil
 	}
 }
+
+// UpdateValidatorMinimumCommission updates the commission rate for all
+// validators to be at least the new min commission rate, and sets the minimum
+// commission rate in the staking params.
+func UpdateValidatorMinimumCommission(
+	ctx sdk.Context,
+	app App,
+) {
+	resultCount := make(map[stakingtypes.BondStatus]int)
+
+	// Iterate over *all* validators including inactive
+	app.stakingKeeper.IterateValidators(
+		ctx,
+		func(index int64, validator stakingtypes.ValidatorI) (stop bool) {
+			// Skip if validator commission is already >= 5%
+			if validator.GetCommission().GTE(ValidatorMinimumCommission) {
+				return false
+			}
+
+			val, ok := validator.(stakingtypes.Validator)
+			if !ok {
+				panic("expected stakingtypes.Validator")
```

Comment:
do we need to panic inline? could be useful to iterate through as many as can be upgraded and collecting a complete list of all invalid ones to make for fewer debug cycles if this unexpected condition is encountered?

Reply:
Don't think it's really needed, the iterator unmarshals 1 type `types.Validator` before sending it to this fn, so if it panics once in the tests it will be the same for all of them